XenAdmin/Wizards/HAWizard.cs

```diff
@@ -108,7 +104,32 @@ protected override void UpdateWizardContent(XenTabPage senderPage)
             {
                 xenTabPageHaFinish.HeartbeatSrName = xenTabPageChooseSR.SelectedHeartbeatSR.Name;
                 xenTabPageHaFinish.Ntol = xenTabPageAssignPriorities.Ntol;
-                xenTabPageHaFinish.RestartPriorities = xenTabPageAssignPriorities.getCurrentSettings().Values;
+
+                int alwaysRestartHighPriority = 0, alwaysRestart = 0, bestEffort = 0, doNotRestart = 0;
+                foreach (VM.HA_Restart_Priority priority in xenTabPageAssignPriorities.getCurrentSettings().Values)

```

Comment:
Thanks for demoing @kc284.

I think "xenTabPageAssignPriorities.getCurrentSettings()" could take long, because that method iterates through the datagridview from the previous page (and casts each row).

I would suggest we change this to have the previous page to calculate this on page leave and then from this page we could just access the result Dictionary what could be just a field in that class. That way the previous page will wait a bit before leaving as it can have a large grid with a lot of data - then this page will load immediately.